### PR TITLE
Integrations: Gradient[Embeddings,LLM] - sdk-upgrade

### PR DESCRIPTION
**Pull request by michaelfeil**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Potential breaking change for `GradientEmbeddings`: New miniumum version of `pip install "gradientai>=1.4.0"` is required. As `gradientai` is not linked with a specific version in `pyproject.toml` this would lead to a breaking change, if the miniumum version is not  `pip install "gradientai>=1.4.0"`. Note: It would be awesome if that could be managed, aka `pip install llama-index[gradient]`. @mhaligowski 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests - for the async
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [not relevant] I have made corresponding changes to the documentation
- [not relevant] I have added Google Colab support for the newly added notebooks.
- [raises importerror / warning ] My changes generate no new warnings 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by michaelfeil** [comment]
@mhaligowski Can you give this a check before I mark it as ready?

**Comment by michaelfeil** [comment]
This is ready.

File: llama_index/embeddings/gradient.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "Represent this sentence for searching relevant passages:"
 )
 
-GRADIENT_MAX_BATCH_SIZE = 100
+GRADIENT_EMBED_BATCH_SIZE = 32_768
```
Comment (hatianzhang) [inline llama_index/embeddings/gradient.py:20]: is this a valid int number

Comment (michaelfeil) [reply llama_index/embeddings/gradient.py:20]: Yes! 

I'll patch it with, to make it easier to validate with mypy. 

```python
GRADIENT_EMBED_BATCH_SIZE: int = 32_768
```


```
Python 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> type(32_768)
<class 'int'>
```

**Review by hatianzhang — APPROVED** [review]

**Pull request merged**